lib/Echidna/Solidity.hs

```diff
@@ -98,6 +100,8 @@ data SolConf = SolConf { _contractAddr    :: Addr             -- ^ Contract addr
                        , _initialize      :: Maybe FilePath   -- ^ Initialize world with Etheno txns
                        , _multiAbi        :: Bool             -- ^ Whether or not to use the multi-abi mode
                        , _checkAsserts    :: Bool             -- ^ Test if we can cause assertions to fail
+                       , _ignoreMethods   :: [String]         -- ^ List of method to avoid calling during a campaign
```

Comment:
What about having something like this here instead:
```haskell
data MethodFilter = Whitelist NonEmpty | Blacklist NonEmpty
_methodFilter :: Maybe MethodFilter
```
and parse the config to this type, having a bit stronger semantics on type level.

Reply:
If I understand correctly, this type should allow json files with either whitelist or blacklist keywords in a config file. If you include both of them, the JSON parsing in Haskell should fail? It seems to be this JSON library was not designed for this kind of semantic parsing, where some keywords are excluded if certain other keyword are present.

Reply:
single JSON object makes this super nice